airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py

```diff
@@ -68,25 +69,30 @@ def get_matching_files(self, globs: List[str], prefix: Optional[str], logger: lo
         total_n_keys = 0
 
         try:
-            if prefixes:
-                for prefix in prefixes:
-                    for remote_file in self._page(s3, globs, self.config.bucket, prefix, seen, logger):
-                        total_n_keys += 1
-                        yield remote_file
-            else:
-                for remote_file in self._page(s3, globs, self.config.bucket, None, seen, logger):
+            for current_prefix in (prefixes if prefixes else [None]):
```

Comment:
Why is this change needed?

Reply:
I simplified the code to avoid duplication